lib/media/segment_index.js

```diff
@@ -497,8 +498,10 @@ shaka.media.SegmentIterator = class {
         'Please use SegmentIndex.getIteratorForTime instead of seek().');
 
     const iter = this.segmentIndex_.getIteratorForTime(time);
-    this.currentPosition_ = iter.currentPosition_;
-    this.currentPartialPosition_ = iter.currentPartialPosition_;
+    if (iter) {
```

Comment:
Will do.. 

The only use of this seek() function that I could find is in the unit tests, and it doesn't seem like this would be accessible for direct use by applications. Can we just remove it?

Reply:
Ultimately, yes, we will remove it.  But for backward compatibility reasons, we can't remove it until Shaka Player v4.  It's exported to the application, and may be used by private plugins.  So for now, we need to make sure it still works.

Reply:
done